api4/resolver_channel.go

```diff
@@ -29,6 +29,31 @@ func (ch *channel) Team(ctx context.Context) (*model.Team, error) {
 	return getGraphQLTeam(ctx, ch.TeamId)
 }
 
+// match with api4.getChannelStats
+func (ch *channel) Stats(ctx context.Context) (*model.ChannelStats, error) {
+	c, err := getCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
All resolvers need to include the auth checks as well:

```go
if !c.App.SessionHasPermissionToChannel(*c.AppContext.Session(), c.Params.ChannelId, model.PermissionReadChannel) {
		c.SetPermissionError(model.PermissionReadChannel)
		return
	}
```

Because this is a graph, every node can be reached from multiple places. So we cannot rely on parent node's auth check.